src/operators/skipUntil.ts

```diff
@@ -19,8 +19,9 @@ class SkipUntilOperator<T, R> implements Operator<T, R> {
 class SkipUntilSubscriber<T> extends Subscriber<T> {
   private notificationSubscriber: NotificationSubscriber<any> = null;
 
-  constructor(destination: Subscriber<T>, private notifier: Observable<any>) {
-    super(destination);
+  constructor(public destination: Subscriber<T>,
+              private notifier: Observable<any>) {
+    super(null);
```

Comment:
@staltz there's a bit of [secret sauce](https://github.com/ReactiveX/RxJS/blob/9d90c753e2a29cd375a49610e3f7bc1963e10d3b/src/Subscriber.ts#L40-L50) in that constructor that is used for setting up the unsubscription chain. I'm having a hard time just from the code change discerning what moving away from `super(destination)` is fixing. Can you describe the issue in more detail?